go/cmd/dolt/commands/sqlserver/server.go

```
@@ -261,12 +278,47 @@ func getDbStates(ctx context.Context, mrEnv env.MultiRepoEnv, dbs []dsqle.Databa
 		}
 
 		head := dEnv.RepoStateReader().CWBHeadSpec()
-		headCommit, err := dEnv.DoltDB.Resolve(ctx, head, dEnv.RepoStateReader().CWBHeadRef())
+
+		var ws *doltdb.WorkingSet
+		var headCommit *doltdb.Commit
+		var err error
+		if _, val, ok := sql.SystemVariables.GetGlobal(DoltDefaultBranchKey); ok {
+			if refStr, ok := val.(string); ok && ref.IsRef(refStr) {
```

Better to throw an error here if this isn't a valid ref string

---

Encapsulate this whole chunk of logic in a method